src/testRunner/unittests/tsbuild.ts

```diff
@@ -394,32 +395,27 @@ namespace ts {
     }
 
     function loadProjectFromDisk(root: string): vfs.FileSystem {
-        const fs = new vfs.FileSystem(/*ignoreCase*/ false, { time });
-        const rootPath = resolvePath(__dirname, root);
-        loadFsMirror(fs, rootPath, "/src");
-        fs.mkdirpSync("/lib");
-        const libs = ["es5", "dom", "webworker.importscripts", "scripthost"];
-        for (const lib of libs) {
-            const content = Harness.IO.readFile(combinePaths(Harness.libFolder, `lib.${lib}.d.ts`));
-            if (content === undefined) {
-                throw new Error(`Failed to read lib ${lib}`);
-            }
-            fs.writeFileSync(`/lib/lib.${lib}.d.ts`, content);
-        }
-        fs.writeFileSync("/lib/lib.d.ts", Harness.IO.readFile(combinePaths(Harness.libFolder, "lib.d.ts"))!);
-        fs.meta.set("defaultLibLocation", "/lib");
+        const resolver = vfs.createResolver(Harness.IO);
+        const fs = new vfs.FileSystem(/*ignoreCase*/ true, {
+            files: {
+                ["/lib"]: new vfs.Mount(vpath.resolve(Harness.IO.getWorkspaceRoot(), "built/local"), patchResolver(resolver)),
+                ["/src"]: new vfs.Mount(vpath.resolve(Harness.IO.getWorkspaceRoot(), root), resolver)
+            },
+            cwd: "/",
+            meta: { defaultLibLocation: "/lib" },
+            time
+        });
         fs.makeReadonly();
         return fs;
-    }
-
-    function loadFsMirror(vfs: vfs.FileSystem, localRoot: string, virtualRoot: string) {
-        vfs.mkdirpSync(virtualRoot);
-        for (const path of Harness.IO.readDirectory(localRoot)) {
-            const file = getBaseFileName(path);
-            vfs.writeFileSync(virtualRoot + "/" + file, Harness.IO.readFile(localRoot + "/" + file)!);
-        }
-        for (const dir of Harness.IO.getDirectories(localRoot)) {
-            loadFsMirror(vfs, localRoot + "/" + dir, virtualRoot + "/" + dir);
+        function patchResolver(resolver: vfs.FileSystemResolver): vfs.FileSystemResolver {

```

Comment:
Is this absolutely necessary? The mounted files are lazily loaded now, right? And only diffs are in the output? What still depends on only these lib files being findable?